### PR TITLE
Treatment Legend

### PR DESCRIPTION
**Pull request by lastminutediorama**
TODO:
- decide if we're not using a backend lookup endpoint anymore, and if this should just pull its data from a local definition
- I put this in an sg-modal, but for elements that fit inside the map component and inherit placement, etc, maybe those deserve their own components.

Screenshot 2024-09-24 at 9 58 49 PM

### CODE REVIEW

File: src/interface/src/styleguide/treatments-legend/treatments-legend.component.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+})
+export class TreatmentsLegendComponent {
+  constructor() {}
+
+  openState = false;
+  @Input() singleTreatments: PrescriptionSingleAction[] = [];
+  @Input() sequences: number[] = [1, 2, 3, 4, 5, 6, 7, 8]; // TODO: make dynamic
+
+  loadAvailableTreatments(): void {
+    // TODO: read this from local types instead of lookups?
```
Comment (lastminutediorama) [inline src/interface/src/styleguide/treatments-legend/treatments-legend.component.ts:41]: @pflopez - when we originally discussed this component, I think we were planning to get our full list of prescription types from the backend. 
Now it seems that `prescriptions.ts` might serve as the FE source of truth. Any opinions on how this component ought to be populated now?

Comment (pflopez) [reply src/interface/src/styleguide/treatments-legend/treatments-legend.component.ts:41]: i think we could use it directly from prescriptions.ts, however if we go that route, maybe this component needs to live on the app / treatments code and not on the styleguide; unless you can come up with a general "legend" component, and a concrete implementation on app/treatments that does the mapping, which im not sure is worth the effort

Comment (pflopez) [reply src/interface/src/styleguide/treatments-legend/treatments-legend.component.ts:41]: I say we keep this component on treatments, and maybe invest the effort on the styleguide side for the modal itself which (which you suggested already on another pr :D )

Comment (lastminutediorama) [reply src/interface/src/styleguide/treatments-legend/treatments-legend.component.ts:41]: I agree, I suspect we'll never be using this anywhere outside of treatments. Styleguide is a convenient place to build/test UI components, but there's no reason this needs to live there.

**Comment by lastminutediorama** [comment]
I trimmed down some vertical gaps, but I'm wondering if this ought to be in a dialog component, after all. I think the legend ought to respond to sizing from the map component, and I'm not sure mat-dialog will let us do that.

File: src/interface/angular.json [hunk 2]
```diff
@@ -153,6 +153,15 @@
         "storybook": {
           "builder": "@storybook/angular:start-storybook",
           "options": {
+            "assets": [
+              "src/favicon.ico",
+              "src/assets",
+              {
+                "glob": "**/*",
```
Comment (lastminutediorama) [inline src/interface/angular.json:160]: glob here, so storybook can use our ng assets folder

**Comment by pflopez** [comment]
Looking great and the collapse/expand animation looks awesome.
One detail- the X at the bottom, should that close the modal (and turn off the view layers setting) ?

File: src/interface/src/app/treatments/treatment-legend/treatment-legend.component.html [hunk 3]
```diff
@@ -1 +1,50 @@
...
+        <div class="expander-heading-content">
+          <mat-icon class="material-symbols-outlined expander-heading-icon"
+            >repeat</mat-icon
+          >Treatment Sequencing
+        </div>
+      </mat-expansion-panel-header>
+      <div class="legend-section">
+        <div class="legend-row" *ngFor="let seq of sequences">
+          <sg-sequence-icon [sequenceNum]="seq"></sg-sequence-icon>
+          <div class="legend-text">{{ 'Sequence' + seq }}</div>
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatment-legend/treatment-legend.component.html:45]: probably want a space between "sequence" and the number? Although not sure if this is going to change with the change on the `SEQUENCE_ACTIONS` we discussed yesterday

**Comment by lastminutediorama** [comment]
> One detail- the X at the ~bottom~ top, should that close the modal (and turn off the view layers setting) ?
I think that's correct, but I'm chatting with UX about it, I'll update when I hear back.

**Review by pflopez — APPROVED** [review]

**Pull request merged**